packages/native/src/__tests__/useUnhandledLinking.test.tsx

```diff
@@ -0,0 +1,373 @@
+      </NavigationContent>
+    );
+  });
+
+  const OuterStack = createStackNavigator();
+  const Stack = createStackNavigator();
+
+  const TestScreen = ({ route, signOut }: any): any => (
+    <>
+      <Text>{route.name}</Text>
+      <Button title="sign out" onPress={signOut} />
+    </>
+  );
+  const SignInScreen = ({ route, signIn }: any): any => {
+    const { handleLastLinking } = useUnhandledLinking();
+    return (
+      <>
+        <Text>{route.name}</Text>
+        <Button
+          title="sign in"
+          onPress={() => {
+            handleLastLinking();
+            signIn();
+          }}
+        />
+      </>
+    );
+  };
+
+  const linking = {
+    prefixes: ['rn://'],
+    config: {
+      screens: {
+        Outer: {
+          path: 'outer',
+          screens: {
+            Home: 'home',
+            Profile: 'profile',
+            Outer: 'outer',
+          },
+        },
+      },
+    },
+    getInitialURL() {
+      return 'rn://outer/profile';
+    },
+  };
+
+  const navigation = createNavigationContainerRef<ParamListBase>();
+
+  const App = () => {
+    const [isSignedIn, setSignedIn] = React.useState(false);
+
+    return (
+      <NavigationContainer ref={navigation} linking={linking}>
+        <OuterStack.Navigator>
+          <OuterStack.Screen name="Outer">
+            {() => (
+              <Stack.Navigator>
+                {isSignedIn ? (
+                  <>
+                    <Stack.Screen name="Home">
+                      {(props) => (
+                        <TestScreen
+                          signOut={() => act(() => setSignedIn(false))}
+                          {...props}
+                        />
+                      )}
+                    </Stack.Screen>
+                    <Stack.Screen name="Profile">
+                      {(props) => (
+                        <TestScreen
+                          signOut={() => act(() => setSignedIn(false))}
+                          {...props}
+                        />
+                      )}
+                    </Stack.Screen>
+                  </>
+                ) : (
+                  <Stack.Screen name="SignIn">
+                    {(props) => (
+                      <SignInScreen
+                        signIn={() => act(() => setSignedIn(true))}
+                        {...props}
+                      />
+                    )}
+                  </Stack.Screen>
+                )}
+              </Stack.Navigator>
+            )}
+          </OuterStack.Screen>
+        </OuterStack.Navigator>
+      </NavigationContainer>
+    );
+  };
+
+  const { queryByText } = await waitFor(() => render(<App />));
+
+  expect(queryByText('SignIn')).not.toBeNull();
+
+  fireEvent.press(queryByText(/sign in/i));
+
+  expect(queryByText('SignIn')).toBeNull();
+
+  expect(queryByText('Profile')).not.toBeNull();
+});
+
+it('should schedule a state to be handled on conditional linking in nested stack', async () => {
```

Comment:
For tests, I use the pattern "does something" instead of "should do something" since it's shorter and means the same thing.

e.g. - `it('schedules a state....`